From 7fa268f6b30dc1f011f8fc95a216ae06683ba0b9 Mon Sep 17 00:00:00 2001
From: Jan Paul Batrina <jpmbatrina01@gmail.com>
Date: Mon, 5 Apr 2021 16:09:05 +0800
Subject: [PATCH] When there is no selection, disable selection-only search

This ensures that the search won't be stuck after the first search.
This can happen when a document is reloaded or when the selection-only
search is manuallyt enabled without actual selection.
---
 src/search/katesearchbar.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git src/search/katesearchbar.cpp src/search/katesearchbar.cpp
index f38f0a34..92f36020 100644
--- src/search/katesearchbar.cpp
+++ src/search/katesearchbar.cpp
@@ -162,6 +162,11 @@ KateSearchBar::KateSearchBar(bool initAsPower, KTextEditor::ViewPrivate *view, K
     connect(docUndoManager, &KateUndoManager::redoStart, this, setSelectionChangedByUndoRedo);
     connect(docUndoManager, &KateUndoManager::redoEnd, this, unsetSelectionChangedByUndoRedo);
 
+    // When document is reloaded, disable selection-only search so that the search won't be stuck after the first search
+    connect(view->doc(), &KTextEditor::Document::reloaded, this, [this]() {
+        setSelectionOnly(false);
+    });
+
     // init match attribute
     Attribute::Ptr mouseInAttribute(new Attribute());
     mouseInAttribute->setFontBold(true);
@@ -543,6 +548,7 @@ bool KateSearchBar::findOrReplace(SearchDirection searchDirection, const QString
         }
     } else {
         // No selection
+        setSelectionOnly(false);
         const Cursor cursorPos = m_view->cursorPosition();
         if (searchDirection == SearchForward) {
             inputRange.setRange(cursorPos, m_view->document()->documentEnd());
-- 
GitLab

